include/uapi/sound/sof/header.h

```diff
@@ -22,7 +22,11 @@ struct sof_abi_hdr {
 	__u32 type;		/**< component specific type */
 	__u32 size;		/**< size in bytes of data excl. this struct */
 	__u32 abi;		/**< SOF ABI version */
-	__u32 reserved[4];	/**< reserved for future use */
+	__u8 ipc_version_priv;	/** 0: SOF_IPC, 1: SOF_INTEL_IPC4 */
```

Comment:
As we changed the sof_abi_hdr structure, we need update the SOF ABI version, and update the SOF_ABI_PATCH to 1? @ranj063

Reply:
now this is tricky isnt it? this is for the purpose ABI 4, so we should update the ABI 3's minor. I prefer creating a new structure for IPC4 instead of updating this struct to use the reserved fields